File: go/raindrops/raindrops.go
package raindrops

import "fmt"

func Convert(input int) string {
	// the factors we are interested in
	var factors = []int{ 3, 5, 7 }

	result := ""
	for _, factor := range factors {
		// do we have a clean divide ?
		if input%factor == 0 {
			switch factor {
				case 3:
					result += "Pling"
				case 5:
					result += "Plang"
				case 7:
					result += "Plong"
			}
		}
	}

	// do we have any matches ?
	if result == "" {
		result = fmt.Sprintf("%d", input)
	}

	return result
}

File: python/raindrops/raindrops.py
# compared to the last solution we only check for the keys we are interested in : 3, 5, 7
# this makes things way easier.

def raindrops(number):
    my_map = {3: 'Pling', 5: 'Plang', 7: 'Plong'}
    res = ''.join([my_map[f] for f in [x for x in my_map.keys() if number % x == 0]])
    return res or str(number)

